core_info.c

```diff
@@ -601,6 +603,13 @@ static core_info_t *core_info_find_internal(
          continue;
       if (string_is_equal(path_basename(info->path), core_path_basename))
          return info;
+
+      if (libretro_pos)
+      {
+         size_t core_name_length = libretro_pos - core_path_basename;
+         if (strncmp(path_basename(info->path), core_path_basename, core_name_length) == 0)
```

Comment:
- We don't need to call `path_basename(info->path)` twice - this can be done once at the start

- Similarly, we don't need to do `size_t core_name_length = libretro_pos - core_path_basename;` on each iteration - it can be done once at the start

Reply:
Fixed.